packages/ai-jsx/src/lib/tts/tts.tsx

```diff
@@ -0,0 +1,360 @@
+'use client';
+
+/**
+ * Defines a function that can be used to build a URL for a text-to-speech
+ * service. This URL will be used to retrieve an audio file that can be
+ * played by an HTML5 audio element.
+ */
+export type BuildUrlFunction = (provider: string, voice: string, rate: number, text: string) => string;
+
+/**
+ * Defines a function that can be used to retrieve an ephemeral access token
+ * for us with a text-to-speech service. Typically, this will be a
+ * fetch request to a server that will return an opaque token.
+ */
+export type GetTokenFunction = (provider: string) => Promise<string>;
+
+/**
+ * Defines a base class for text-to-speech services. This class provides
+ * a common interface for text-to-speech services, as well as some basic
+ * infrastructure for playing audio using the HTML5 audio element.
+ */
+export class TextToSpeechBase {
+  protected audio: HTMLAudioElement;
+  /**
+   * The time when the play() method was first called.
```

Comment:
Is this epoch time or time since the page load began? (The latter is what you get from the web `performance` API.)

Reply:
monotonic, e.g., performance.now

Reply:
Ok, I think that would be good to put in the comment.